test/optimizations/autoLocalAccess/use-before-def-bug.chpl

```diff
@@ -0,0 +1,6 @@
+// from issue #23653
```

Comment:
Could you make this more self-contained by adding a sentence summarizing the issue here?